data_structures/quad_tree/quad_tree.dart

```diff
@@ -0,0 +1,219 @@
+// Author: Jerold Albertson
+// Profile: https://github.com/jerold
+// Algorithm: https://en.wikipedia.org/wiki/Quadtree
+
+import 'dart:math';
+import 'package:test/test.dart';
+
+// defaults should almost never be used, tune the quad tree to fit your problem
+int default_max_depth = 1000;
+int default_max_items = 100;
+
+// names reflect a coordinate system where values increase as one goes left or down
+int _upperLeftIndex = 0;
+int _upperRightIndex = 1;
+int _lowerLeftIndex = 2;
+int _lowerRightIndex = 3;
```

Comment:
I thought about making those variables `const`, but i think it's fine.

Reply:
Thanks for the review @Parowicz. Yeah, these should have been const, luckily I at least made them private. It would have been bad had they been exposed like the defaults above. I'll cut a quick followup pr to fix that. Happy Hacktoberfest!